cdap-common/src/main/java/io/cdap/cdap/common/metadata/QueryTerm.java

```diff
@@ -78,6 +80,28 @@ public QueryTerm(String term, Qualifier qualifier, SearchType searchType, Compar
     this.qualifier = qualifier;
     this.searchType = searchType;
     this.comparison = comparison;
+    this.date = null;
+  }
+
+  /**
+   * Constructs a QueryTerm using search term, qualifying, type, and numeric information.
+   *
+   * @param term the search term
+   * @param qualifier the qualifying information {@link Qualifier}
+   * @param searchType the search method information {@link SearchType}
+   * @param comparison the comparison operator for the numeric value {@link Comparison}
+   * @param number the number which can be associated with a date or a numeric value, depending on the search type
+   */
+  public QueryTerm(String term, Qualifier qualifier, SearchType searchType, Comparison comparison, Long number) {
+    this.term = term;
+    this.qualifier = qualifier;
+    this.searchType = searchType;
+    this.comparison = comparison;
+    if (this.searchType.equals(SearchType.DATE)) {
+      this.date = number;
+    } else {
+      this.date = null;
```

Comment:
you can always initialize date with null and overwrite the value if needed.

Reply:
hm, when I try to do that I get an error that date is already initialized, and the issue is coming from date being a final field. I can't figure out how to overwrite it, but please let me know if there is a way!

Reply:
May be we can do this? It is less verbose.  `this.date = this.searchType.equals(SearchType.DATE) ? number : null;`

Reply:
yes that works, thanks!